io/plugins/eu.esdihumboldt.hale.io.gml.test/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter2Test.groovy

```diff
@@ -381,4 +381,72 @@ class StreamGmlWriter2Test {
 
 		result
 	}
+
+	@Test
+	void testNumbers() throws Exception {
+		double doubleNumber = 123456789.123456789
+		double expected = java.lang.Double.parseDouble(doubleNumber.toString())
+
+		testNumber("doubleSimpleNumber", doubleNumber, expected)
+
+		testNumber("integerNumber", 12345, null)
+		testNumber("doubleNumber", 12345.123456789, null)
+		testNumber("floatNumber", 12345.6789F, null)
+		testNumber("longNumber", 1234567890123456789L, null)
+		testNumber("shortNumber", 12345 as short, null)
+		testNumber("decimalNumber", new BigDecimal("1234567890.123456789"), null)
+
+		testNumber("decimalNumber", new BigDecimal(doubleNumber), null);
+
+		doubleNumber = 1.23456789123456789E8
+		testNumber("decimalNumber", new BigDecimal(doubleNumber), null);
```

Comment:
I think an additional test would be great where the scientific notation is passed into the `BigDecimal` constructor as a string.